traits/trait_dict_object.py

```diff
@@ -364,13 +369,15 @@ def pop(self, key, default_value=None):
                 if the key was present.
 
         """
-        if key in self:
-            removed = {key: self[key]}
-            result = super().pop(key)
-            self.notifiy(removed=removed)
-            return result
-        else:
-            return default_value
+        if value is Undefined or key in self:
+            removed = super().pop(key)
```

Comment:
This method can now raise a `KeyError`, need to update docstring.

Reply:
This one I updated the description copied from Python, which already mentions KeyError. It does not hurt to add another KeyError section...